internal/b/blueprints.go

```diff
@@ -206,6 +228,18 @@ func normalizeApplicationService(as ApplicationService) (ApplicationService, err
 	as.HSToken = hex.EncodeToString(hsToken)
 	as.ASToken = hex.EncodeToString(asToken)
 
+	if as.URL == "" {
+		// Since, we're just checking and not reserving the port, we could
+		// potentially run into an issue where the port is no longer available when
+		// we actually try to bind to it later on
+		port, err := getFreePort()
+		if err != nil {
+			return as, err
+		}
+
+		as.URL = fmt.Sprintf("http://%s:%d", HostnameRunningComplement, port)
```

Comment:
Just checking, do we even care about having a random port for the application service?

Previously I just had it run on port `9111` but changed it to a random port in https://github.com/matrix-org/complement/pull/221/commits/06935a72aa3a5fa00fd458949c63ed993294e85d

Reply:
Yes, tests are run concurrently so we can't bind to the same port all the time.